### DISCUSSION TITLE
Running on Rpi 3B

### DISCUSSION THREAD

**Discussion by rymiller11** [ep1cman/unifi-protect-backup#125] [Ideas]
Thought I'd share for other noobs trying to get this up and running.  I was never able to get UPB running with docker but instead am running it directly from the command line (i'll be starting it as a service as soon as I'm happy there are no issues).

The first couple of times I started UPB it would crash and the only message showing in the output was "killed".  I increased the verbosity and still never found out why it would crash.  I also found out that the video file that was in progress of downloading to my NAS would be corrupted and not viewable.  So, whatever happened it was not graceful.

Here's what I did to fix it - or at least it seems to be working better.  Using Conky, i saw that my swap file was maxed out (100%).  That seemed odd so I increased it from the default 100MiB to 1000MiB and UPB has not crashed since.  I see that I'm using at most 20% of the swap file (about 175MiB) and it seems to stick there.  If UPB indeed needs 175 and the default is 100 then this may be the answer.

Anyway, hope this helps others trying to get UPB working with cheap hardware.

Here's my setup:

- UDMSE with Protect running
- RPi 3B running primarily Unifi Protect Backup (+Conky, VNC, and usual software)
- Netgear ReadyNAS 102 (a very low powered beginner NAS that I've had for years and it keeps on running)

I think if I had an RPi with 1GbE it would copy files much faster.  The NAS is 1 GbE but it may be bottlenecked because of the slow/old CPU.

Cheers

**Comment by ep1cman** [comment]
How much RAM do you have on the device? UPB needs to be able able to keep video clips in RAM to prevent SD card wear. It could be that large videos are filling the RAM, this would cause a sudden crash with no error message as you are seeing. You could try reducing the size of the download queue

**Reply by rymiller11** [reply]
I have a GB of memory.  Why it would get killed is a mystery because it appears to be plenty of memory, especially with a swap file of 100.

What I also noticed was that one of the files that it crashed on was one of the larger files.  Could there be multiple files in memory at the same time?  If so, that sounds like the culprit.

![image](https://github.com/ep1cman/unifi-protect-backup/assets/89878121/bed432f0-4a2f-4c64-a3f7-a289a5d22605)

**Reply by ep1cman** [reply]
There is a command line option (and corresponding env car) "--download-buffer-size", this sets how large the download buffer should be and it will download as many videos as it can to keep that buffer full. By default it is set to 512mb so it's feasible that the rest of the system could be using ovee 512mb and causing the crash. Try reducing it to 256mb and see if that helps. It could be a particularly large video download that causes the issue, we can't tell how large they will be till after they are downloaded sadly

You want to avoid swap if you can help it other this application will constantly be writing to the SD card and kill it prematurely.
